src/itemsList/ItemsList.tsx

```diff
@@ -31,33 +32,23 @@ class ItemsList extends Component<ItemsListProps> {
 
   render() {
     const { children, className, withSeparators, keyGenerator, ...otherProps } = this.props
-    let separator = false
+
     return (
-      <ul
-        className={cc([
-          'kirk-items-list',
-          className,
-          { 'kirk-items-list--withSeparators': withSeparators },
-        ])}
-        {...otherProps}
-      >
+      <ul className={cc(['kirk-items-list', className])} {...otherProps}>
         {children.map((item, index) => {
           let child = null
-          if (item.type === ItemsListDivider && !withSeparators) {
-            separator = true
-          } else if (item.type !== ItemsListDivider && item.type !== undefined) {
+          const isLast = children.length === index + 1
+          let hasSeparator =
+            !isLast && (children[index + 1].type === ItemsListDivider || withSeparators)
+
+          if (item.type !== ItemsListDivider && item.type !== undefined) {
             child = (
-              <li
-                className={cc([
-                  'kirk-items-list-item',
-                  { 'kirk-items-list-item--withSeparator': separator },
-                ])}
-                key={keyGenerator(index)}
-              >
+              <li className={cc(['kirk-items-list-item'])} key={keyGenerator(index)}>
                 {item}
+                {hasSeparator && <Divider />}
               </li>
             )
-            separator = false
+            hasSeparator = false
```

Comment:
This hasSeparator is never used (it's reset by the next "let hasSeparator =") in the next loop iteration.
Am I missing something or something is missing or can we just remove that line safely ?